File: problems/704. Binary Search/Solution.java
class Solution {
    public int search(int[] nums, int target) {
        if (nums.length == 1) return (nums[0] == target) ? 0 : -1;
        int value = recurse(nums, 0, nums.length - 1, target);
        return value;
    }

    private int recurse(int[] nums, int start, int end, int target) {
        if (start == end) return (nums[start] == target) ? start : -1;
        int pivot = (end - start) / 2;
        int leftVal = recurse(nums, start, start + pivot, target);
        int rightVal = recurse(nums, start + pivot + 1, end, target);
        return (leftVal + rightVal != -2) ? ((leftVal == -1) ? rightVal : leftVal) : -1;
    }
}

File: problems/704. Binary Search/Solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if len(nums) == 1: return 0 if nums[0] == target else -1
        value = self.recurse(nums, 0, len(nums) - 1, target)
        return value

    def recurse(self, nums: List[int], start: int, end: int, target: int) -> int:
        if start == end: return start if nums[start] == target else -1
        pivot = (end - start) // 2
        leftVal = self.recurse(nums, start, start + pivot, target)
        rightVal = self.recurse(nums, start + pivot + 1, end, target)
        if leftVal != -1: return leftVal
        if rightVal != -1: return rightVal
        return -1